aws/resource_aws_appsync_function_test.go

```diff
@@ -0,0 +1,314 @@
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAwsAppsyncFunctionExists(resourceName, &config),
+					resource.TestCheckResourceAttr(resourceName, "description", "test description 1"),
+				),
+			},
+			{
+				Config: testAccAWSAppsyncFunctionConfigDescription(rName1, rName2, testAccGetRegion(), "test description 2"),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAwsAppsyncFunctionExists(resourceName, &config),
+					resource.TestCheckResourceAttr(resourceName, "description", "test description 2"),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccAwsAppsyncFunction_responseMappingTemplate(t *testing.T) {
+	rName1 := fmt.Sprintf("tfacctest%d", acctest.RandInt())
+	rName2 := fmt.Sprintf("tfexample%s", acctest.RandString(8))
+	resourceName := "aws_appsync_function.test"
+	var config appsync.FunctionConfiguration
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAwsAppsyncFunctionDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSAppsyncFunctionConfigResponseMappingTemplate(rName1, rName2, testAccGetRegion()),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAwsAppsyncFunctionExists(resourceName, &config),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccAwsAppsyncFunction_disappears(t *testing.T) {
+	rName1 := fmt.Sprintf("tfacctest%d", acctest.RandInt())
+	rName2 := fmt.Sprintf("tfexample%s", acctest.RandString(8))
+	resourceName := "aws_appsync_function.test"
+	var config appsync.FunctionConfiguration
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAwsAppsyncFunctionDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSAppsyncFunctionConfig(rName1, rName2, testAccGetRegion()),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAwsAppsyncFunctionExists(resourceName, &config),
+					testAccCheckAwsAppsyncFunctionDisappears(resourceName),
+				),
+				ExpectNonEmptyPlan: true,
+			},
+		},
+	})
+}
+
+func testAccCheckAwsAppsyncFunctionDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*AWSClient).appsyncconn
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_appsync_function" {
+			continue
+		}
+
+		apiID, functionID, err := decodeAppsyncFunctionID(rs.Primary.ID)
+		if err != nil {
+			return err
+		}
+
+		input := &appsync.GetFunctionInput{
+			ApiId:      aws.String(apiID),
+			FunctionId: aws.String(functionID),
+		}
+
+		_, err = conn.GetFunction(input)
+		if err != nil {
+			if isAWSErr(err, appsync.ErrCodeNotFoundException, "") {
+				return nil
```

Comment:
We should fix this in the future:

```suggestion
				continue
```